caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -767,26 +868,40 @@ def encode(
             convert_to_numpy = False
 
         input_was_string = False
-        if isinstance(sentences, str) or not hasattr(
-            sentences, "__len__"
+        list_of_sentences = sentences
+        if isinstance(list_of_sentences, str) or not isinstance(
+            sentences, Sized
         ):  # Cast an individual sentence to a list with length 1
-            sentences = [sentences]
+            list_of_sentences = [sentences]
             input_was_string = True
 
+        error.type_check_all("<NLP82314994E>", str, sentences=list_of_sentences)
+
         if device is None:
-            device = self._target_device
+            device = self.device
```

Comment:
q - is this guaranteed to be set vs. `self._target_device`? Would this depend on another `.encode()` being called first (like the warmup)?